lib/manageiq/providers/hawkular/entity_mapper.rb

```diff
@@ -0,0 +1,37 @@
+require 'cgi'
+
+module ManageIQ
+  module Providers
+    module Hawkular
+      class EntityMapper
+        def self.map(data)
+          new(data).map
+        end
+
+        def initialize(data)
+          @data = data.dup.symbolize_keys
+        end
+
+        def map
+          feed = @data[:resourceTypePath].match(/;([0-9a-zA-Z.\s%]+)\/rt/)[1]
```

Comment:
I think the right regex for the feed is `/f;([^/])/` (didn't tested it). But I think you should use `CanonicalPath` class of the ruby gem as a helper to extract the feed.

Anyway, this will need to change, because for inventory v4 there are no canonical paths and the feed cannot be inferred from the paths/ids.

Reply:
There's a second branch that I did implement InventoryV4 for GVM. With it, EntityMapper is going to be gone.

https://github.com/cfcosta/manageiq-providers-hawkular/pull/2